pkg/download/id_extraction/id_extraction.go

```diff
@@ -0,0 +1,62 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package id_extraction
+
+import (
+	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2/parameter/value"
+	project "github.com/dynatrace/dynatrace-configuration-as-code/pkg/project/v2"
+	"golang.org/x/exp/maps"
+	"regexp"
+	"strings"
+)
+
+// meIDRegexPattern matching a Dynatrace Monitored Entity ID which consists of a type containing characters and
+// underscores, a dash separator '-' and a 16 char alphanumeric ID
+var meIDRegexPattern = regexp.MustCompile(`[a-zA-Z_]+-[A-Za-z0-9]{16}`)
+
+var uuidRegexPattern = regexp.MustCompile(`[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}`)
+
+// ExtractIDsIntoYAML searches for Dynatrace ID patterns in each given config and extracts them from the config's
+// JSON template, into a YAML parameter. It modifies the given configsPerType map.
+func ExtractIDsIntoYAML(configsPerType project.ConfigsPerType) project.ConfigsPerType {
+	for _, cfgs := range configsPerType {
+		for _, c := range cfgs {
+			ids := meIDRegexPattern.FindAllString(c.Template.Content(), -1)
+			ids = append(ids, uuidRegexPattern.FindAllString(c.Template.Content(), -1)...)
+			ids = deduplicateIDs(ids)
+
+			for _, id := range ids {
+				idKey := strings.ReplaceAll(id, "-", "_") // golang template keys must not contain hyphens
+				paramID := fmt.Sprintf("__EXTRACTED_ID_%s__", idKey)
```

Comment:
Why the prefix? I get that we use the pattern for conversation, but for new properties I would use a simpler `key` only or `id_key`. 
The double underscores are IMO really not necessary.

Reply:
Maybe it would even make sense to differentiate between uuids and meids?
```
parameters:
  uuid_1_2_3_4: 1_2_3_4
  meid_host_1234: host_1234
```

i take it that the prefix was introduced for tools to simply detect the ids as 'monaco found' ids? Otherwise, i would have suggested to get rid of the prefix.

Reply:
You're right, I applied it based on the conversion convention. 
But unlike there, where it visually differentiates keys that existed before to generated ones, all are generated on download. 

I've simplified to to `extractedID_[...]`

Reply:
Works for me

Reply:
just pushed as refactor commit (too lazy to fixup again)